### DISCUSSION TITLE
[Feature Request]: Enforced hop limits on packet relay for repeaters and routers

### DISCUSSION THREAD

**Discussion by noon92** [meshtastic/firmware#3833] [Ideas]
### Platform

NRF52

### Description

As meshes grow larger, the issue of users manually increasing hop limits will become more acute. Users assume that a higher hop limit is by necessity better - and in a perfect world of unlimited network capacity they'd be right. In our world, however, such users can cause major congestion, reducing the usability of the network for all, especially in locales with duty cycle maximums.

Right now, if a node receives a packet with a remaining hop count of 7, it will subtract 1 from that count and relay the packet, even if its own hop limit is set to 3.

Routers and repeaters are infrastructure nodes, and are usually placed in high up positions with good antennas. If such a node relays a packet with a high hop count, it can cause major congestion throughout the mesh over a large area.

I propose a new setting, Enforced Hop Limit, for routers and repeaters. With this setting enabled, if a router receives a packet with a remaining hop count higher than the hop limit, the hop limit will be lowered to the router's hop limit. Alternately, it could be lowered to one below the router's hop limit to simulate if it had been sent with a "proper" hop limit.

This setting could be enabled by admins in particularly congested meshes, especially if hop limit abuse becomes an issue.

**Comment by BillSobel** [comment]
"Alternately, it could be lowered to one below the router's hop limit to simulate if it had been sent with a "proper" hop limit."

I question if the above is useful with multiple peers and detrimental if your only peer is the relay. Logic is this; In a scenario where you know you the only radio you reach is a relay, setting your count to relay_max+1 provides the best network coverage. You get full coverage as if the relay transmitted the message initially. If you have multiple peers, I see why you would want to enforce it to relay_max. 

The router could also do this as an 'if possible' basis meaning if channel utilization is below a threshold transmit the unmodified hop count but if usage goes to high, enforce the count as above.

**Comment by SR7673** [comment]
As far as i understand, if a user sets the hop_limit to 7 and sends a message to a router and then to another node, then the router (having the hop_limit set to the standard of 3) will forward the message with a hop_limit of 3, not of 6.
I tend to set the hop_limit of my Routers (in superior remote sites) to 2. This will have no disadvantage for users having a hop_limit of 3 and there will be no difference between setting the initial hop_limit to 3, 4, 5, 6 or 7.

**Comment by CamFlyerCH** [comment]
Hello @noon92 . I made a similar request and it was also converted to a discussion https://github.com/meshtastic/firmware/discussions/4764 . 
I just found this discussion and I think we agree on a lot of points. I have developed and successfully tested a modification to the firmware. But there is also a lot of work to make it configurable (simple and also adjustable). My issue is, that just as I installed the custom firmware in my router and hoped for improvement, another router was setup by someone else with even better reception, and now I still get all the messages from bad configured nodes. I calculated savings up to 50% less outgoing messages from my router. Since here in Europe we have limitation on sending time, the routers have to drop messages instead of routing them to not use more TX time than allowed.
